### ISSUE TITLE
modify +format-on-save-disabled-modes not work

### ISSUE THREAD

**Issue by ethercflow** [doomemacs/doomemacs#7599]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

I write 
```lisp
(setq +format-on-save-disabled-modes (add-to-list '+format-on-save-disabled-modes 'c-mode))
(setq +format-on-save-disabled-modes (add-to-list '+format-on-save-disabled-modes 'cc-mode))
(setq +format-on-save-disabled-modes (add-to-list '+format-on-save-disabled-modes 'C/*l))
```
on my own `config.el`.

in my own `init.el`, I enabled `(format +onsave)`.

The **expect** is when saving buffer, the c code will not be reformatted

### Current behavior

The **reality** is still reformatted.

When I edit a C file, then save it. the C code still be reformatted.

I run `describe-variable` and make sure the `+format-on-save-disabled-modes` has already changed to:
```
(C/*l cc-mode c-mode sql-mode tex-mode latex-mode org-msg-edit-mode)
```

### Steps to reproduce

1. in init.el

```elisp
:lang
(cc +lsp +tree-sitter)
:editor
(format +onsave)
```

2. in config.el
```elisp
(setq +format-on-save-disabled-modes (add-to-list '+format-on-save-disabled-modes 'c-mode))
(setq +format-on-save-disabled-modes (add-to-list '+format-on-save-disabled-modes 'cc-mode))
(setq +format-on-save-disabled-modes (add-to-list '+format-on-save-disabled-modes 'C/*l))
```

3. open a C project, edit C code file, then save it.

### System Information

https://pastebin.com/R635mPiq

**Comment by EsAu79p** [comment]
I haven't found a solution but I can recreate the behavior somewhat. 
Specifically it happens on the first open of a C code file but stops happening after re-executing major mode.

**Comment by cpg314** [comment]
I can also confirm the issue.
As a temporary fix, on can directly call `apheleia-inhibit` in a hook, e.g.
```elisp
(setq-hook! 'web-mode-hook apheleia-inhibit t)
```

**Comment by lizhuohua** [comment]
I confirm this issue. It only occurs when opening the first project because `Apheleia` has not been loaded yet. And the variable `apheleia-inhibit-functions`, which is a list of functions that prevent `Apheleia` from activating automatically, is initialized only after the package is loaded, as shown here:

https://github.com/doomemacs/doomemacs/blob/ff33ec8f7a89d168ca533612e2562883c89e029f/modules/editor/format/config.el#L46-L50

As a temporary fix, I added the following to my `config.el`:

```elisp
(use-package! apheleia
  :config
  (add-to-list 'apheleia-inhibit-functions #'+format-maybe-inhibit-h))
```

**Comment by ethercflow** [comment]
Thanks @lizhuohua, I've tried your temporary fix, I works now. BTW, I found another problem:

When I type
```c
int a()
{ // press enter
```
It will gen:

```c
int a() {
// first line will be 8 space instead of tab
// this line is tab
}
```
Do you have any idea to reslove this?

**Comment by lizhuohua** [comment]
@ethercflow Sorry, I don't use c-mode or cc-mode, so I don't know how to resolve your problem. I think you can create a new issue to see if others can help.

**Comment by ethercflow** [comment]
> @ethercflow Sorry, I don't use c-mode or cc-mode, so I don't know how to resolve your problem. I think you can create a new issue to see if others can help.

@lizhuohua Got it, thanks!  :)

**Comment by ethercflow** [comment]
BTW, I find the best way to auto format the c code is using .clang-format.

**Comment by hlissner** [comment]
This issue should've been resolved two months ago in 8072762de8dd9bf43651a8676642081d92027782. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
